docs/reusing-typecheck-results.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Reusing typecheck results
+category: Compiler Internals
+categoryindex: 200
+index: 42
+---
+
+# Reusing typecheck results between compiler and tooling runs
+
+Caching and reusing typecheck results between compiler and tooling runs will be an optimization technique aimed at improving the performance and efficiency of the F# development experience.
+
+### Why is it important
+
+- **Performance**. Recomputing type information for large projects is time-consuming. By caching the results, subsequent runs can skip this step, leading to faster builds.
+
+- **IDE responsiveness**. IDEs can provide faster IntelliSense, code navigation, and other features if they can quickly access cached type information.
+
+- **Error reduction**. Consistent typechecking results help avoid scenarios where slight differences between runs could cause different errors or warnings to be produced.
+
+### General considerations
+
+- The technique will be implemented separately for the compiler and tooling, for better granularity and trackability.
```

Comment:
In addition to that, tooling and compiler can be using different version of compiler service, and technically results might be incompatible

Reply:
Yeah this will be a big hurdle (repeat analyzers discussions)

Reply:
Yes, we need to keep the track of the compiler version, noted.